### ISSUE TITLE
Register Form: Save new  Organisation location data

### ISSUE THREAD

**Issue by trungjc** [Kademi/kademi-dev#6603]
Register form which creates new organisation should be save organisation location data to org detail

Reference to https://github.com/Kademi/fuel/issues/16

**Comment by bradmac** [comment]
I think the first part of this is to add options to the registerForm component to give an option for creating a new org
- update the settings page to have an option for "Allow creating new org'
- and another option for whether users should enter org address fields when creating a new org
- and an option for a fixed country
- if the option to allow creating new orgs is enabled, then users can either select an existing org, or add a new one (see suggested UI below)
- if the user is adding a new org, and the option to enter address is enabled, also show address fields (addressLine1, line2, city, postcode, and country if a fixed country is not selected in settings)

Here is a suggested UI for the select or create UI from another site;

Selection mode:
![image](https://user-images.githubusercontent.com/826458/51722086-1bdcbb00-20b9-11e9-8f3b-e2e7506a677d.png)

Create mode:
![image](https://user-images.githubusercontent.com/826458/51722110-30b94e80-20b9-11e9-9c10-69d104c03bb0.png)

**Comment by anhnt** [comment]
![image](https://user-images.githubusercontent.com/735555/51724834-bfb27080-2091-11e9-9f18-a2a5a0cfc55d.png)
![image](https://user-images.githubusercontent.com/735555/51724836-c6d97e80-2091-11e9-870b-e1810e502249.png)

UI done

**Comment by anhnt** [comment]
![image](https://user-images.githubusercontent.com/735555/51724887-086a2980-2092-11e9-901c-4a69ffbe0d8c.png)

**Comment by bradmac** [comment]
I'm re-implementing the rego form handler in KSignup

I'm doing this in 2 different methods
- doImmediateRego: for open groups
- doRegoApplication: for admin approval groups

**Comment by bradmac** [comment]
I've done doImmediateRego, but not tested yet. Should be backwards compatible with existing form

**Comment by bradmac** [comment]
@trungjc please make these changes to this component in the editor:

-    checkboxes and labels should be on the same row in settings, otherwise its too hard to read
-    it seems like option selections arent used in edit mode, the form always displays with minimal fields

**Comment by trungjc** [comment]
![screenshot_1](https://user-images.githubusercontent.com/704946/51822629-80915300-230f-11e9-982f-8b446614b231.png)

@bradmac is ready for https://github.com/Kademi/fuel/issues/16 ?

**Comment by bradmac** [comment]
@trungjc there's still a lot of excess space between fields, which makes this form hard to use:
![image](https://user-images.githubusercontent.com/826458/51866303-8dfd1a80-23ad-11e9-8d32-7fcf181bc379.png)

Can you please tighten that up?

**Comment by bradmac** [comment]
@trungjc in edit mode, only one organisation input should be shown

![image](https://user-images.githubusercontent.com/826458/51882379-8ce6e000-23e3-11e9-9aa7-5cf5778e05d1.png)

**Comment by bradmac** [comment]
The new KSignup JS based controller is mostly working now. 

Need to update pendingAccountsPortlet and pendingApplicationsComponent to work with the new extended org fields stuff

**Comment by trungjc** [comment]
![screenshot_4](https://user-images.githubusercontent.com/704946/51888006-3029fc00-23c8-11e9-8862-ad5214378524.png)
![screenshot_5](https://user-images.githubusercontent.com/704946/51888007-3029fc00-23c8-11e9-9594-e2b358f0cbfa.png)

Update setting page

**Comment by Uafinder** [comment]
1. When you created a new account and open form for rego, there are all options enabled
![image](https://user-images.githubusercontent.com/1945230/52050267-ac534980-2558-11e9-922f-3e56f5e6cf57.png)
http://vladtest56j.admin.kademi-ci.co/websites/vladtest56jweb/version1?editTemplate=theme/apps/signup/register
![image](https://user-images.githubusercontent.com/1945230/52050287-c0974680-2558-11e9-8b3d-8448a2a25685.png)
http://vladtest56jweb_vladtest56j.kademi-ci.co/participants/signup
When you submit current form it will give you an NPE error
![image](https://user-images.githubusercontent.com/1945230/52050325-de64ab80-2558-11e9-8e42-f4eb9dbaadd1.png)

2. When there are no organisations created yet, create and select organisation parameters doesn't appear

3. Enable this create org doesn't cause it to appear on custom page
![image](https://user-images.githubusercontent.com/1945230/52050624-d22d1e00-2559-11e9-860f-a5143e0a0aba.png)
http://vladtest56j.admin.kademi-ci.co/pages/vladtest56jweb/version1/?contenteditor=page2.html
http://vladtest56jweb_vladtest56j.kademi-ci.co/page2.html
![image](https://user-images.githubusercontent.com/1945230/52050648-e4a75780-2559-11e9-8701-6879c0ef9c27.png)

Well, it looks buggy for me

**Comment by bradmac** [comment]
> When you created a new account and open form for rego, there are all options enabled

Thats ok. Looks a bit messy but this way users can see whats available

**Comment by bradmac** [comment]
> Enable this create org doesn't cause it to appear on custom page

**Comment by bradmac** [comment]
@Uafinder over to you

**Comment by bradmac** [comment]
Requires signup-lib 1.3.1

**Comment by bradmac** [comment]
Improved support for org select custom fields.

Now you can choose the org type by adding the orgtype code to the options list:
![image](https://user-images.githubusercontent.com/826458/52172476-f6486380-27d4-11e9-97ea-4c839614fd45.png)

Currently only support a single org type

Requires KSignup 1.1.0 / signup-lib 1.3.2 / entityFinder-lib 1.0.3

**Comment by Uafinder** [comment]
By default,
when using standard signup form
new user doesn't get membership in rootorg and doesn't fill name and surname fields
as well as password
![image](https://user-images.githubusercontent.com/1945230/52259143-5e5c8e00-292a-11e9-8ac3-cab49f6dce5a.png)
http://vladtest56lweb_vladtest56l.kademi-ci.co/learners/signup

When you approve membership, username is not disappear from list
![image](https://user-images.githubusercontent.com/1945230/52259226-a11e6600-292a-11e9-8d33-0661d67ae6bc.png)

Error 500 when using custom form
http://vladtest56lweb-vladtest56l.kademi-ci.co/pagegroup1.html
![image](https://user-images.githubusercontent.com/1945230/52259584-e2fbdc00-292b-11e9-8800-d4541a68d2ba.png)
![image](https://user-images.githubusercontent.com/1945230/52259621-10e12080-292c-11e9-8385-111ade8da398.png)

**Comment by bradmac** [comment]
> When you approve membership, username is not disappear from list

This is correct and by design, so the user can click through to their profile. Note that it does change though.

**Comment by bradmac** [comment]
> new user doesn't get membership in rootorg and doesn't fill name and surname fields

Fixed. Requires signup-lib 1.3.3 / KSignup 1.1.0

**Comment by bradmac** [comment]
> Error 500 when using custom form

Increased column size in CI DB, so should be fixed now

**Comment by Uafinder** [comment]
1. Error JS in console on custom component when use Org search box
![image](https://user-images.githubusercontent.com/1945230/52467641-f064d080-2b8e-11e9-9035-d003cb16405a.png)
http://vladtest56mmmweb-vladtest56mmm.kademi-ci.co/page1signup.html

2. Creating org via signup form doesn't add it to index
![image](https://user-images.githubusercontent.com/1945230/52467989-f3ac8c00-2b8f-11e9-9619-c028449ebae3.png)
http://vladtest56mmm.admin.kademi-ci.co/manageUsers/18757352/#summary-tab

http://vladtest56mmm.admin.kademi-ci.co/manageSearch/org/
![image](https://user-images.githubusercontent.com/1945230/52468015-0f179700-2b90-11e9-83f7-1f60ba8f00c0.png)

3. 

> 
> 
> > When you created a new account and open form for rego, there are all options enabled
> 
> Thats ok. Looks a bit messy but this way users can see whats available

Nah, i meant that all options are enabled by default when you open form for edition, but all this options doesn't appear on frontend. Both regular and custom component.
![image](https://user-images.githubusercontent.com/1945230/52468129-6d447a00-2b90-11e9-83c1-13216cd64419.png)
![image](https://user-images.githubusercontent.com/1945230/52468164-81887700-2b90-11e9-8d37-436538c7d78a.png)
http://vladtest56mmmweb-vladtest56mmm.kademi-ci.co/contenteditor?fileName=page1signup.html

**Comment by bradmac** [comment]
>     1. Error JS in console on custom component when use Org search box

Thats because the "customers" group has not been made available to this website, which you can see on the group management page:

![image](https://user-images.githubusercontent.com/826458/52538479-f0a3dc80-2dd7-11e9-98e7-80656836470e.png)

**Comment by bradmac** [comment]
>     2. Creating org via signup form doesn't add it to index

Fixed, requires platform update

**Comment by bradmac** [comment]
> all options are enabled by default when you open form for edition, but all this options doesn't appear on frontend

@trungjc can you please resolve this? just needs correct defaulting when values are missing/empty

**Comment by trungjc** [comment]
![screenshot_12](https://user-images.githubusercontent.com/704946/52551297-6460dc00-2e0e-11e9-84d1-cb816e8b65f4.png)
![screenshot_13](https://user-images.githubusercontent.com/704946/52551298-64f97280-2e0e-11e9-963a-2c188353fd96.png)

User have to add sign-up group to [Website group access](http://vladtest56mmm.admin.kademi-ci.co/websites/vladtest56mmmweb/version1/#panel_groups-tab), the extra fields and opt-ins will be showed

**Comment by bradmac** [comment]
Need to be able to configure which group to use for org extra fields

**Comment by bradmac** [comment]
> User have to add sign-up group to [Website group access](http://vladtest56mmm.admin.kademi-ci.co/websites/vladtest56mmmweb/version1/#panel_groups-tab), the extra fields and opt-ins will be showed

This is correct, but it would be nice to have some kind of help text or warning. I'll look at that

**Comment by anhnt** [comment]
requires admin-lib 1.4.0 and entityFinder-lib 1.0.5 to show org extra fields on manage profile page

**Comment by bradmac** [comment]
@anhnt looks like we have a z-index issue with the entity-finder, can you look into that?

![image](https://user-images.githubusercontent.com/826458/52599443-79487880-2ebd-11e9-9019-85da9395e648.png)

**Comment by bradmac** [comment]
I've removed the code in GMA processing which created a membership for org selector fields. This was never applied consistently, and i think isnt needed. Will be quite complex to do it properly so remove for now.

**Comment by anhnt** [comment]
>looks like we have a z-index issue with the entity-finder, can you look into that?

Fixed in latest entityFinder lib

**Comment by anhnt** [comment]
just fixed several issues regarding rego form

**Comment by Uafinder** [comment]
Selecting Opt-in doesn't cause adding user to that group
![image](https://user-images.githubusercontent.com/1945230/52940402-282dfe00-336f-11e9-9fe2-d2e4c522b483.png)
![image](https://user-images.githubusercontent.com/1945230/52940429-38de7400-336f-11e9-976c-6608d8ec8bdf.png)
http://vladtest56sweb_vladtest56s.kademi-ci.co/ecommerce-users/signup

**Comment by anhnt** [comment]
@Uafinder please do not check on `Enable multipart form (file upload)` and try again

**Comment by Uafinder** [comment]
so please make sure this option is deselected by default

**Comment by Uafinder** [comment]
Also, when similar org names appears (e.g. more than 4-5 choices available) org search show that no orgs find
![image](https://user-images.githubusercontent.com/1945230/52942118-06cf1100-3373-11e9-8f34-03741c0ce367.png)
http://vladtest56sweb_vladtest56s.kademi-ci.co/ecommerce-users/signup

**Comment by anhnt** [comment]
>Also, when similar org names appears (e.g. more than 4-5 choices available) org search show that no orgs find

@bradmac this seems works if i enter whole title or org id (ie `org1`)
can you please check

**Comment by bradmac** [comment]
> Also, when similar org names appears (e.g. more than 4-5 choices available) org search show that no orgs find

@anhnt  I've reproduced this locally with 5 orgs named dealer1 ... dealer5

As @Uafinder says, when i enter 3 letters nothing appears. However, i've checked the ajax response and it contains the 5 orgs as expected.

When there are only 2 orgs (dealer1 and dealer2) they appear correctly.

So it seems like there is some option to only display results if there are less then about 5.

This is the ajax response with 5 orgs;
![image](https://user-images.githubusercontent.com/826458/54080818-65c5eb80-435d-11e9-8c50-a2391920c527.png)

No orgs shown:
![image](https://user-images.githubusercontent.com/826458/54080827-a02f8880-435d-11e9-807f-e9695a88a133.png)
